### PR TITLE
Vehicle status

### PR DESCRIPTION
**Pull request by mohitrawat789**

### CODE REVIEW

**Comment by mohitrawat789** [comment]
This work is Regarding Close Shift .

File: Parking/Forms/MPS.cs [hunk 1]
```diff
@@ -56,12 +52,23 @@ private void btn_PendingVehicle_Click(object sender, EventArgs e)
...
             }
-        }                
+            catch (Exception s)
+            {
+                MessageBox.Show("Error Loading Shift Data", null, MessageBoxButtons.OK, MessageBoxIcon.Error);
+                return;
+            }            
+        }
+
+        private void btn_VehicleStatus_Click(object sender, EventArgs e)
```
[the remark this answers is not in the document]
Comment (mohitrawat789) [reply Parking/Forms/MPS.cs]: Yes as the user would other wise feel like nothing is happening when he clicks **Close Shift** Button and when he sees the error he can check the network connectivity or restart the application.

File: Parking/Forms/MPS.cs [hunk 2]
```diff
@@ -56,12 +52,23 @@ private void btn_PendingVehicle_Click(object sender, EventArgs e)
 
         private void btn_CloseShift_Click(object sender, EventArgs e)
         {
-            if (!_shiftClosed)
+            try
             {
-                _parkingDatabaseFactory.SaveMPSUserShiftExit(_mpsUserEntryRecordIdentifier);
-                this.Close();
-                _shiftClosed = true;
+                var result = _parkingDatabaseFactory.GetShiftData("2018-10-29 23:28:09.000");
```
[the remark this answers is not in the document]
Comment (mohitrawat789) [reply Parking/Forms/MPS.cs]: Removed. I used it for test data.

File: Parking.Interfaces/IParkingDatabaseFactory.cs [hunk 3]
```diff
@@ -25,8 +25,6 @@ public interface IParkingDatabaseFactory
 
         DataTable GetPendingVehicles();
 
-        object SaveMPSUserShiftEntry(string deviceId, string userId, string inDateTime);
-
-        void SaveMPSUserShiftExit(string mpsUserEntryRecordIdentifier);
+        Tuple<int, int> GetShiftData(string entryTime);
```
[the remark this answers is not in the document]
Comment (mohitrawat789) [reply Parking.Interfaces/IParkingDatabaseFactory.cs]: Done

File: Parking.Database/CommandFactory/ParkingDatabaseFactory.cs [hunk 4]
```diff
@@ -187,30 +183,26 @@ public DataTable GetPendingVehicles()
...
+                    count = (int)result.Rows[0]["RecordCount"];
+                    collection = (object)result.Rows[0]["TotalCollection"] == DBNull.Value ? 0 :(int)result.Rows[0]["TotalCollection"];
+
+                }
+                return Tuple.Create(count, collection);
             }
             catch (Exception exception)
             {
-                FileLogger.Log($"MPS User shift Exit details could not be saved in database as : {exception.Message}");
+                FileLogger.Log($"Shift Data Could not be retrieved from database as : {exception.Message}");
```
[the remark this answers is not in the document]
Comment (mohitrawat789) [reply Parking.Database/CommandFactory/ParkingDatabaseFactory.cs]: Done

File: Parking.Database/CommandFactory/ParkingDatabaseFactory.cs [hunk 5]
```diff
@@ -187,30 +183,26 @@ public DataTable GetPendingVehicles()
...
-                var insertQuery = string.Format(queries["InsertMPSDeviceShift"], deviceId, userId, inDateTime);                
-                return sqlDataAccess.ExecuteNonQueryReturnsIdentity(insertQuery);
-            }
-            catch (Exception exception)
-            {
-                FileLogger.Log($"MPS User shift Enrty details could not be saved in database as : {exception.Message}");
-                throw;
-            }
-        }
+                var getQuery = string.Format(queries["GetShiftData"], entryTime);
```
[the remark this answers is not in the document]
Comment (mohitrawat789) [reply Parking.Database/CommandFactory/ParkingDatabaseFactory.cs]: Yes

File: Parking.Database/CommandFactory/ParkingDatabaseFactory.cs [hunk 6]
```diff
@@ -187,30 +183,26 @@ public DataTable GetPendingVehicles()
             }
         }
 
-        public object SaveMPSUserShiftEntry(string deviceId, string userId, string inDateTime)
+        public Tuple<int, int> GetShiftData(string entryTime)
         {
```
[the remark this answers is not in the document]
Comment (mohitrawat789) [reply Parking.Database/CommandFactory/ParkingDatabaseFactory.cs]: Done

**Comment by ankitit2015** [comment]
Merged

**Pull request merged**